chore: Revert MSE reset when crossing boundaries

joeyparrish opened the pull request:
This reverts commit b9c634519284b32cdd71039987d531e44f5e0b7c (#8240) and commit 369916489e3b1017b814a9b51e88c2e07150b827 (#8156).

This change is made via git revert on each commit, and git commit --amend to squash the two reverts together.

matvp91 commented:
For reference, I've reused the content workarounds test asset from [here](https://github.com/shaka-project/shaka-player/blob/main/test/media/content_workarounds_integration.js) and it seems that one started failing as of today. I quickly ran the content workarounds tests from v4.13.7 and I'm experiencing the same error:

```
 {
   "severity": 2,
   "category": 1,
   "code": 1001,
   "data": [
     "https://drm-widevine-licensing.axtest.net/AcquireLicense",
     403,
     "",
     {
       "content-length": "0",
       "x-axdrm-errormessage": "License denied. The DRM message references the ID of an unknown communication key.",
       "x-axdrm-identity": "Axinom DRM - Widevine API",
       "x-axdrm-server": "widevine-testing-7887bd76d5-fs6bz",
       "x-axdrm-version": "6.27.1"
     },
     2,
     "https://drm-widevine-licensing.axtest.net/AcquireLicense"
   ],
   "handled": false,
   "message": "Shaka Error 1001",
   "stack": "Error: Shaka Error 1001"
 }
```

Unfortunately, I don't have an asset available with both plain and encrypted periods for public testing.

joeyparrish commented:
😭

joeyparrish commented:
You're right.  It's the asset, not the test or the feature.

> Unfortunately, I don't have an asset available with both plain and encrypted periods for public testing. If you know of an available one, I'd be happy to update the PR's.

It's easy to make one.  Just stitch together content from MPDs into a multi-period one.  For example, Sintel or Tears of Steel from the Shaka content.  No need to even duplicate the segments, just create a new MPD that references the old segments.  I'll close this PR and put up another that changes the content URL instead.

Thanks!

joeyparrish commented:
I took https://storage.googleapis.com/shaka-demo-assets/sintel-widevine/dash.mpd (encrypted) and https://storage.googleapis.com/shaka-demo-assets/sintel/dash.mpd (clear).  I removed mediaPresentationDuration from MPD and added it as duration on Period instead.  Then I added a BaseURL element to each Period so they pointed to the original folders for segments.  Finally, I combined Periods and gave each Period a unique ID.

Here are the results:

 - https://storage.googleapis.com/shaka-demo-assets/sintel-mixed-encryption/enc-clear-enc.mpd
 - https://storage.googleapis.com/shaka-demo-assets/sintel-mixed-encryption/clear-enc-clear.mpd

matvp91 commented:
Perfect, thank you. I'll get to it first thing in the morning.

joeyparrish commented:
Took a bit longer to extract segments, key IDs, keys, and build PlayReady PSSHs for the version in the tests.  See https://github.com/shaka-project/shaka-player/pull/8244
